R/utils.R

```diff
@@ -327,4 +335,66 @@ valid_seq <- function(params) {
   } else {
     return(FALSE)
   }
-}
\ No newline at end of file
+}
+
+resolve_prefix <- function(prefix, environment_var) {
+  if (!(is.null(prefix))) {
+    assertthat::assert_that(is.character(prefix))
+    
+    return(prefix)
+  } else {
+    prefix_env <- Sys.getenv(environment_var)
+    if (prefix_env != "") {
+      return(prefix_env)
+    } else {
+      return("/")
+    }
+  }
+}
+
+get_package_mapping <- function(script_path, url_package, dependencies) {
+  package_map <- vapply(dependencies, function(x) {
+    if (x$name %in% c('react', 'react-dom')) {
+      x$name <- 'dash_renderer'
+    }
+    dep_path <- paste(x$src$file, 
+                      x$script,
+                      sep = "/")
+  
+    # remove double slash and replace with single if present
```

Comment:
when is there a double slash vs a single slash?

Reply:
This problem will go away when I :hocho: the use of `htmltools` for generating the script tags. It seems to want to insert a leading slash in the tag, and I couldn't figure out how to remove it.

A far better solution is to do what you suggested earlier today, and manually generate the script tag as in Python. I'll take care of that and :hocho: these lines when they're no longer needed.